prov/rxm/src/rxm_ep.c

```diff
@@ -775,7 +771,7 @@ rxm_ep_format_tx_res_lightweight(struct rxm_ep *rxm_ep, struct rxm_conn *rxm_con
 	(*tx_buf)->pkt.hdr.tag = tag;
 
 	if (flags & FI_REMOTE_CQ_DATA) {
-		(*tx_buf)->pkt.hdr.flags = OFI_REMOTE_CQ_DATA;
+		(*tx_buf)->pkt.hdr.flags = FI_REMOTE_CQ_DATA;
```

Comment:
This is a protocol change.

The provider also seems to assume that communicating peers have the same Endianness.  Is this true?

Reply:
Good catch.
We need to ensure that OP version of wire protocol matches each others.

@shefty Is it OK to detect endianness and fail if they don't match? We ensure that CM data (that is sent in `fi_connect`/`fi_accept()`) is sent in big-endian (network) byte order. And then, we can check endianess of peers and if they don't match we fail connection establishment procedure,
This approach requires having the same endianness on both machines, but it avoids overhead at fast-path rather than we have to convert form host to network byte order every time we need to send data.

Reply:
I'm fine with this approach.  I don't think mixed endianness is likely, so failing is reasonable.  If it does become an issue, we can add patches to address it, such that the conversion is only made when needed.